src/hooks/useStreamingPaymentAmountsLeft.ts

```diff
@@ -0,0 +1,71 @@
+import { BigNumber } from 'ethers';
+
+import { type StreamingPayment } from '~gql';
+
+const useStreamingPaymentAmountsLeft = (
+  streamingPayment: StreamingPayment | null | undefined,
+  currentTimestamp: number,
+) => {
+  if (!streamingPayment)
+    return {
+      amountsClaimedToDate: {},
+      amountsAvailableToClaim: {},
+    };
+
+  const amountsClaimedToDate: { [tokenAddress: string]: BigNumber } =
+    streamingPayment.claims
+      ? streamingPayment.claims.reduce((amounts, claim) => {
+          let newAmount = BigNumber.from(claim.amount);
+
+          if (amounts[claim.tokenAddress]) {
+            newAmount = BigNumber.from(claim.amount).add(
+              amounts[claim.tokenAddress],
+            );
+          }
+
+          return { ...amounts, [claim.tokenAddress]: newAmount };
+        }, {})
+      : {};
+
+  const amountsAvailableToClaim = streamingPayment.payouts
+    ? streamingPayment.payouts.reduce((amounts, payout) => {
+        let newAmount = BigNumber.from(0);
+        if (streamingPayment.startTime >= currentTimestamp) {
+          return { amounts, [payout.tokenAddress]: newAmount };
+        }
+
+        const durationToClaim =
+          Math.min(currentTimestamp, streamingPayment.endTime) -
+          streamingPayment.startTime;
+
+        if (!durationToClaim) {
+          return { amounts, [payout.tokenAddress]: newAmount };
+        }
+
+        const amountAvailableSinceStart = BigNumber.from(payout.amount)
+          .mul(
+            BigNumber.from(durationToClaim)
+              .mul(BigNumber.from(10).pow(18))
+              .div(BigNumber.from(streamingPayment.interval)),
+          )
+          .div(BigNumber.from(10).pow(18));
```

Comment:
I was going to make a comment about using hardcoded 18 as token decimals but upon further reading further on this I suppose it's a good representation of how solidity does multiplication - https://github.com/dapphub/ds-math?tab=readme-ov-file#wmul. Nice work figuring this out 👍